src/main/java/net/greghaines/jesque/worker/WorkerImpl.java

```diff
@@ -529,21 +533,40 @@ protected void process(final Job job, final String curQueue) {
             final Object instance = this.jobFactory.materializeJob(job);
             final Object result = execute(job, curQueue, instance);
             success(job, instance, result, curQueue);
+        } catch (DontPerformException e) {
+            try {
+                Thread.sleep(dontPerformSleepTime);
```

Comment:
The Ruby you linked to sleeps the worker, too. I just don't think telling the worker thread to sleep while there's a good chance there are other jobs to process is a good pattern.